cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/stream/hbase/HBaseStreamConsumerStateStoreFactory.java

```diff
@@ -54,7 +59,9 @@ public synchronized StreamConsumerStateStore create(StreamConfig streamConfig) t
     TableId streamStateStoreTableId = StreamUtils.getStateStoreTableId(namespace);
     TableId hbaseTableId = tableUtil.createHTableId(new NamespaceId(streamStateStoreTableId.getNamespace()),
                                                     streamStateStoreTableId.getTableName());
-    try (HBaseAdmin admin = new HBaseAdmin(hConf)) {
+    try (HBaseDDLExecutor executor = new HBaseDDLExecutorFactory(cConf, hConf).get();
+         // TODO should tableExists be part of HBaseDDLExecutor??
+         HBaseAdmin admin = new HBaseAdmin(hConf)) {
       if (!tableUtil.tableExists(admin, hbaseTableId)) {
```

Comment:
Since `tableExist` method is not part of the `HBaseDDLExecutor`, however we still want to reuse the `admin`, should `HBaseDDLExecutor` have `getAdmin` method?